ipmininet/examples/bgp_full_config.py

```diff
@@ -0,0 +1,102 @@
+from ipmininet.iptopo import IPTopo
+from ipmininet.router.config import RouterConfig, BGP, ebgp_session, set_local_pref, set_med, set_rr, new_access_list, set_community, new_community_list, AF_INET6
+
+
+class BGPTopoFull(IPTopo):
+	"""This topology is composed of two AS connected in dual homing with different local pref"""
+	def build(self, *args, **kwargs):
+		"""
+	TODO
+		   +----------+                                   +--------+
+					  |                                   |
+		 AS1          |                  AS2              |        AS3
+					  |                                   |
+					  |                                   |
+	+-------+   eBGP  |  +-------+     iBGP    +-------+  |  eBGP   +-------+
+	| as1r1 +------------+ as2r1 +-------------+ as2r2 +------------+ as3r1 |
+	+-------+         |  +-------+             +-------+  |         +-------+
+					  |                                   |
+					  |                                   |
+					  |                                   |
+		 +------------+                                   +--------+
+		"""
+		# Add all routers
+		as1r1 = self.addRouter('as1r1')
+		as1r1.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as1r2 = self.addRouter('as1r2')
+		as1r2.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as1r3 = self.addRouter('as1r3')
+		as1r3.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as1r4 = self.addRouter('as1r4')
+		as1r4.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as1r5 = self.addRouter('as1r5')
+		as1r5.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as1r6 = self.addRouter('as1r6')
+		as1r6.addDaemon(BGP, address_families=(
+            AF_INET6(redistribute=('connected',)),))
+		as4r1 = self.addRouter('as4r1')
+		as4r1.addDaemon(BGP, address_families=(AF_INET6(networks=('dead:beef::/32',)),))
+		as4r2 = self.addRouter('as4r2')
+		as4r2.addDaemon(BGP, address_families=(AF_INET6(networks=('dead:beef::/32',)),))
+		as4h1 = self.addHost("as4h1")
+
+		# Add Links
+		self.addLink(as1r1, as1r6, params1={"ip": ("fd00:1:1::1/48",)},
+					 params2={"ip": ("fd00:1:1::2/48",)})
```

Comment:
it was easier to debug with each interface set rather than checking all the time the ip address of the routers while using traceroute

Reply:
Does subnet always allocate the first router with the address::1 and the second to the address::2?

Reply:
> it was easier to debug with each interface set rather than checking all the time the ip address of the routers while using traceroute

You should favor the clarity of the example code over anything else.
After all, this is mostly on this basis that new users will try to understand how to use your work.
Therefore, you should modify the test code so that it gets back the interface address (i.e., ```node.intf('<interface-name>').ip6```) whenever you need it instead of lengthening a code with static addressing.

> Does subnet always allocate the first router with the address::1 and the second to the address::2?

Yes, addresses are allocated in the order of the list of routers